compiler/crates/relay-compiler/src/build_project/mod.rs

```diff
@@ -205,6 +206,14 @@ pub fn build_programs(
     Ok((programs, Arc::new(source_hashes)))
 }
 
+fn is_relay_file_path(language: &TypegenLanguage, path: &PathBuf) -> bool {
+    match (path.to_str(), &language) {
```

Comment:
do we need to convert this `to_str()`

can you use this: https://doc.rust-lang.org/std/path/struct.Path.html#method.ends_with?

Reply:
Yes, I think so, because it seems ends_with there only checks for entire chunks of the path. So that ends_with would mean the entire last chunk (the full file name in this case), which isn't what we're after. I tried using that function first. But maybe there's a better alternative still.